Updating Legacy JSON.md

```diff
@@ -0,0 +1,269 @@
+
+`ident` is exclusively used in `type: material`, `type: MOD_INFO`, `type: profession`, `type: scenario`, `type: skill_display_type` and `type: skill`. If you see `ident` used anywhere else, immediately replace it with `id`. If you see `id` or `abstract`, used in any of the above `type`, immediately replace with `ident`. Using the wrong identifier can cause major errors with the game.
+
+`abstract` can only be used on `type: TOOL`, `type: GENERIC`, `type: GUN`, `type: COMESTIBLE`, `type: BOOK`, `type: AMMO`, `type: PET_ARMOR`, `type: vehicle_part`, `type: BIONIC_ITEM`, `type: ARMOR`, `type: TOOLMOD`, `type: ENGINE`, `type: MONSTER`, `type: uncraft`, and `type: overmap_terrain`. 
+
+`abstract` creates a pseudo-item that only exists to be copied from and is discarded after JSON loading is complete. If you see it used outside of these fields, replace with `ident` or `id` as appropriate. For debugging purposes, it is preferable to not use `abstract` on types other than `type: overmap_terrain`. It can cause cascading errors that are impossible to find due to lack of item id and presence in the game. See [JSON Inheritance](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_INHERITANCE.md) for more information on how to use abstract.
+
+`id` can (and perhaps should) be used anywhere `abstract` can be. It is used almost anywhere and doesn't need updating. Check out [JSON Info](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_INFO.md) for more information on where to use `id`. 
+
+
+# Ammo Type
+It is possible to specify the `key: value` pair `ammo: string` under both `type: GUN`, and `type: AMMO`. However, under `type: GUN`, the `key: value` pair should actually be `ammo: [ string ]`. 
+For reference, here is what it should look like under `type: GUN`:
+```JSON
+"ammo": [ "300blk" ]
+```
+And what it should look like under `type: AMMO`:
+```JSON
+"ammo": "300blk"
+```
+
+It is practically impossible to replace all at once, due to the similarities between `type: GUN` and `type: AMMO`. However, if you exclude all `type: AMMO` from the search (manually or otherwise), this will work:
+```regex
+"ammo": "([^\s]+)"
+
+"ammo": [ "$1" ]
+```
+
+
+# Artifacts
+Artifacts are currently undergoing a massive change. Nothing too important has changed yet, so this is simply a placeholder section.
+
+
+# barrel_length
+Very recently, the `barrel_length` `key` for `type: GUN` has been replaced by `barrel_volume`. Fortunately this is a rather easy fix:
+```regex
+"barrel_length":
+
+"barrel_volume":
+```
+
+Less recently, at some point, `barrel_length` switched `value` from a `number` to a `string`. The modern JSON looks like this:
+```JSON
+"barrel_volume": "750 ml"
+```
+While the obsolete JSON looks like this:
+```JSON
+"barrel_length": 1
+```
+As I don't currently know the conversion from `number` to `string`, I can't recommend a change. 
```

Comment:
```suggestion
Multiply the number stated here by 250 and add 'ml' so for example "barrel_length": 3 becomes "barrel_length": "750ml"
```
You can see the example here:
https://github.com/CleverRaven/Cataclysm-DDA/pull/37738

Reply:
Cool. Thanks man!